registry/application/services/service_publisher_service.py

```diff
@@ -191,18 +191,37 @@ def get_services_for_organization(self, payload):
         search_attribute = payload["s"]
         sort_by = payload["sort_by"].lower()
         order_by = payload["order_by"].lower()
+        filter_parameters_total_count = {  "search_string": search_string,
+                                           "search_attribute": search_attribute if search_attribute in
+                                                                                   ALLOWED_ATTRIBUTES_FOR_SERVICE_SEARCH
+                                           else DEFAULT_ATTRIBUTE_FOR_SERVICE_SEARCH}
+        search_count = ServicePublisherRepository().\
+            get_total_count_of_services_for_organization(self._org_uuid,filter_parameters_total_count)
+        # a = ("a", "b", "c", "d", "e", "f", "g", "h")
+        # x = slice(3, 5)  , so 3 is the offset , which means starting a[3] and 2 elements after (2 is the limit)
+        # which is a[5]
+        # front end will send the offset as 3 and limit as 2 , hence we need adjust
+        # print(a[x]) , ('d', 'e')
+        revised_offset = offset
+        if revised_offset > search_count:
+            revised_offset = search_count
+        revised_limit = offset + limit
```

Comment:
Why are we modifying the limit? This will be the page size which is constant right?

Reply:
thsi is now handled at the query level